### ISSUE TITLE
Accuracy with TimeDistributed(Dense()) models 

### ISSUE THREAD

**Issue by bapla** [fchollet/keras#2301]
It would be great to have a built-in accuracy metric function that ignores paddings. 
I have a sequence model with return_sequences=True, predicting an output for every timestep. As labels need padding as well, metric=['accuracy'] will count the paddings, too.

**Comment by braingineer** [comment]
This one works, but I don't know how to do the `nonzero()` indexing for tensorflow, so i haven't written up a commit yet. 

```python

def categorical_accuracy(y_true, y_pred, mask=None):
    if mask is not None:
        eval_shape = (reduce(mul, K.shape(y_true)[:-1]), K.shape(y_true)[-1])
        y_true_flat = K.reshape(y_true, eval_shape)
        y_pred_flat = K.reshape(y_pred, eval_shape)
        mask_flat = K.flatten(mask).nonzero()[0]  ### how do you do this on tensorflow?
        evalled = K.gather(K.equal(K.argmax(y_true_flat, axis=-1),
                                   K.argmax(y_pred_flat, axis=-1)), 
                           mask_flat)
        return K.mean(K.gather(evalled, good_entries))

    else:
        return K.mean(K.equal(K.argmax(y_true, axis=-1),
                              K.argmax(y_pred, axis=-1)))

```

You have to pass the mask in appropriately as well in training.py (currently they aren't being passed).
